src/systems/odometry_publisher/OdometryPublisher.cc

```diff
@@ -287,13 +292,15 @@ void OdometryPublisher::PreUpdate(const gz::sim::UpdateInfo &_info,
   }
 
   // Create the pose component if it does not exist.
+  //! [createComponent]
```

Comment:
We actually don't need to create the `Pose` component since it's created by default on every `Model` entity, so this code can be removed. We don't have documentation on the default list of components (https://github.com/gazebosim/gz-sim/issues/2306), but we should at least remove counter examples.

Reply:
[9c0b2c4](https://github.com/gazebosim/gz-sim/pull/2219/commits/9c0b2c46ff46e5fd6924bcb7fcf819ba1a2ca0f5)